Jenkinsfile

```diff
@@ -661,6 +665,65 @@ pipeline {
       }
     }
 
+    stage('Deploy Preview') {
+      when {
+        not {
+          branch 'PR-*'
+        }
+      }
+      steps {
+        setGitHubBuildStatus('nuxeo/preview/deploy', 'Deploy nuxeo Preview', 'PENDING')
+        container('maven') {
+          dir('ci/helm/preview') {
+            echo """
+            ----------------------------------------
+            Deploy Preview environment
+            ----------------------------------------"""
+            // first substitute docker image names and versions
+            sh """
+              mv values.yaml values.yaml.tosubst
+              envsubst < values.yaml.tosubst > values.yaml
+            """
+            // second create target namespace (if doesn't exist) and copy secrets to target namespace
+            script {
+              sh(returnStdout: true, script: 'jx -b ns | sed -r "s/^Using namespace \'([^\']+)\'.+\\$/\\1/"')
+              boolean nsExist = sh(returnStatus: true, script: "kubectl get namespace ${NAMESPACE}") == 0
+              String noCommentOpt = '';
+              if (nsExist) {
+                noCommentOpt = '--no-comment'
+              } else {
+                sh "kubectl create namespace ${NAMESPACE}"
+              }
+              sh "kubectl create secret generic preview-docker-cfg --namespace=${NAMESPACE} --from-file=.dockerconfigjson=/home/jenkins/.docker/config.json --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"
+              // third build and deploy the chart
+              // waiting for https://github.com/jenkins-x/jx/issues/5797 to be fixed in order to remove --source-url
+              sh """
+                jx step helm build --verbose
+                mkdir target && helm template . --output-dir target
+                jx preview \
+                  --namespace ${NAMESPACE} \
+                  --verbose \
+                  --source-url=https://github.com/nuxeo/nuxeo \
+                  --preview-health-timeout 15m \
+                  ${noCommentOpt}
+              """
+            }
+          }
+        }
+      }
+      post {
+        always {
+          archiveArtifacts allowEmptyArchive: true, artifacts: '**/requirements.lock, **/charts/*.tgz, **/target/**/*.yaml'
```

Comment:
I'm wondering if we need/want the chart tgz archives, since we have all the YAML files?
@kevinleturc

Reply:
IMO, no we don't.

Reply:
Once things are stable we don't but I would leave it commented so we can uncomment when wanting to debug

Reply:
OK if you think it can be useful, then clearly state in a comment that it can be uncommented for debugging purpose.
By the way, what  is in the archives that is not in the YAML archived files?

Reply:
I removed the tgz. After all if someone wants to play with the yaml he will just have to helm dependency update on the downloaded yaml